core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,352 @@
+  });
+
+  it('should save the given new introduction html', function() {
+    var fakeNewIntroductionHtml = '<p> new intro </p>';
+    simpleEditorManagerService.saveIntroductionHtml(fakeNewIntroductionHtml);
+    expect(stubs.SimpleEditorShimService.saveIntroductionHtml)
+      .toHaveBeenCalledWith(fakeNewIntroductionHtml);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(fakeNewIntroductionHtml);
+  });
+
+  it('should save the customization args', function() {
+    var stateName = 'Introduction';
+    var newCustomizationArgs = {
+      choices: {
+        value: [
+          '<p>Option 3</p>'
+        ]
+      }
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveCustomizationArgs(stateName,
+      newCustomizationArgs);
+    expect(stubs.SimpleEditorShimService.saveCustomizationArgs)
+      .toHaveBeenCalledWith(stateName, newCustomizationArgs);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should save the answer groups', function() {
+    var stateName = 'Introduction';
+    var newAnswerGroups = [{
+      rule_specs: [{
+        rule_type: 'Equals',
+        inputs: {
+          x: 1
+        }
+      }]
+    }];
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveAnswerGroups(stateName, newAnswerGroups);
+    expect(stubs.SimpleEditorShimService.saveAnswerGroups)
+      .toHaveBeenCalledWith(stateName, newAnswerGroups);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should save the default outcome', function() {
+    var stateName = 'Introduction';
+    var newDefaultOutcome = {
+      dest: 'Introduction',
+      feedback: [
+        '<p>One is wrong bcz ofisEmpty many reasons</p>'
+      ]
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveDefaultOutcome(stateName,
+      newDefaultOutcome);
+    expect(stubs.SimpleEditorShimService.saveDefaultOutcome)
+      .toHaveBeenCalledWith(stateName, newDefaultOutcome);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should save the bridge html', function() {
+    var stateName = 'Introduction';
+    var newBridgeHtml = '<p> lets move to next Qsn </p>';
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveBridgeHtml(stateName, newBridgeHtml);
+    expect(stubs.SimpleEditorShimService.saveBridgeHtml)
+      .toHaveBeenCalledWith(questionList.getNextStateName(stateName),
+        newBridgeHtml);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should add new question', function() {
+    var DEFAULT_INTERACTION = {
+      ID: 'MultipleChoiceInput',
+      CUSTOMIZATION_ARGS: {
+        choices: {
+          value: ['<p>Option 1</p>']
+        }
+      }
+    };
+
+    var lastStateName = (questionList.isEmpty() ? SimpleEditorShimService
```

Comment:
Break after '('. Prefer breaking after '?' and ':' rather than breaking at '.', if possible.